src/error/exporterrors.cpp

```diff
@@ -301,247 +276,88 @@ static ostream & printKDBErrors (ostream & os, parse_t & p)
 	   << "			KEY_VALUE, \"the specification of all error codes\", KEY_END)," << endl;
 	for (size_t i = 1; i < p.size (); ++i)
 	{
-		if (p[i]["unused"] == "yes")
-		{
-			continue;
-		}
-
-		os << "		keyNew (\"system/elektra/modules/error/specification/" << i << "\"," << endl
+		os << "		keyNew (\"system/elektra/modules/error/specification/" << p[i]["number"] << "\"," << endl
 		   << "			KEY_END)," << endl
-		   << "		keyNew (\"system/elektra/modules/error/specification/" << i << "/description\"," << endl
-		   << "			KEY_VALUE, \"" << p[i]["description"] << "\", KEY_END)," << endl
-		   << "		keyNew (\"system/elektra/modules/error/specification/" << i << "/severity\"," << endl
-		   << "			KEY_VALUE, \"" << p[i]["severity"] << "\", KEY_END)," << endl
-		   << "		keyNew (\"system/elektra/modules/error/specification/" << i << "/module\"," << endl
-		   << "			KEY_VALUE, \"" << p[i]["module"] << "\", KEY_END)," << endl;
+		   << "		keyNew (\"system/elektra/modules/error/specification/" << p[i]["number"] << "/description\"," << endl
+		   << "			KEY_VALUE, \"" << p[i]["description"] << "\", KEY_END)," << endl;
+		//		   << "		keyNew (\"system/elektra/modules/error/specification/" << p[i]["number"] << "/severity\","
+		//<<
+		// endl
+		//		   << "			KEY_VALUE, \"" << p[i]["severity"] << "\", KEY_END)," << endl
+		//		   << "		keyNew (\"system/elektra/modules/error/specification/" << p[i]["number"] << "/module\"," <<
+		// endl
+		//		   << "			KEY_VALUE, \"" << p[i]["module"] << "\", KEY_END)," << endl;
 	}
 	os << "		KS_END);" << endl << "}" << endl;
 
-	os << "static inline void elektraTriggerWarnings (int nr, Key *parentKey, const char *message)" << endl
-	   << "{" << endl
-	   << "	switch (nr)" << endl
-	   << "	{" << endl;
+	os << "static inline void elektraTriggerWarnings (const char *nr, Key *parentKey, const char *message)" << endl << "{" << endl;
 	for (size_t i = 1; i < p.size (); ++i)
 	{
-		if (p[i]["unused"] == "yes")
-		{
-			continue;
-		}
-
-		if (p[i]["severity"] != "warning") continue;
-		os << "		case " << i << ": ELEKTRA_ADD_WARNING (" << i << ", parentKey, message);" << endl
-		   << "			break;" << endl;
+		os << "	if(strcmp(nr, \"" << p[i]["number"] << "\") == 0)" << endl << "	{" << endl;
```

Comment:
Are we sure that the strcmp is safe? Coding guidelines are violated here: space after if.

Reply:
We must compare a string here, what else would you suggest?

Reply:
strncmp